[all] Replace all Vector2, Vector3, Vector4 by their short name alias Vec2, Vec3

damienmarchal opened the pull request:
The underlying idea is that we should have only one alias for code consistency and to make things clearer.
I didn't deprecates/removed the long names alias but this should be done in a future PR.






______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

damienmarchal committed:
[all] Replace all Vector2, Vector3, Vector4 by their short name alias Vec2, Vec3

The underlying idea is that we should have only one alias for code consistency and
to make things clearer.

I didn't deprecates/removed the long names alias but this should be done in a future PR.

alxbilger reviewed (changes requested):
I agree that the `Vector3` alias must disappear.

The problem is that both aliases are not equivalent:

- `Vector3` = `Vec<3,double>`
- `Vec3` = `Vec<3,SReal>`

I agree that they're the same when `SReal`==`double`, which is the default case. But I don't think we should assume it is always the case.

To be rigorous, here is what I suggest:

- In this PR, which is the first step to clean this mess, I suggest to replace `Vector3` by its true underlying type which is `Vec3d`. It should not affect anything. No conversion is introduced.

- In a second PR, `Vector3` is deprecated.

- Then, we need to check if each individual `Vec3d` should be actually a `Vec3` or not. And yes, it's long, boring and sometimes complex.

I don't see other ways

damienmarchal commented:
I fully agree this is the right (and long) process. 

About the Vec3d to Vec3 move... given that all our code base is in a broken/inconsistant state on that aspect do you think we could give a try to move it directly to Vec3, then locate if (or where) things are getting bad by compiling sofa using REAL=float and fix these parts. 

Side note: I do actually consider that allowing to choose between float/double is a coding style dating from the 2000. In our code base this is causing more troubles (code complexity cost, code maintenance cost, calculus consistency, conversion cost) than the gain we can have , on modern (2020) CPU architectures.

alxbilger commented:
It's not a bad idea to set `SReal` to float. But I would also remove this constructor:
```cpp
    template<typename real2>
    constexpr Vec(const Vec<N, real2>& p) noexcept
    {
        for(Size i=0; i<N; i++)
            this->elems[i] = static_cast<ValueType>(p(i));
    }
```

and probably also this operator:
```cpp
    template<Size M, typename real2>
    constexpr void operator=(const Vec<M,real2>& v) noexcept
    {
        for(Size i=0; i<(N>M?M:N); i++)
            this->elems[i] = (ValueType)v(i);
    }
```

damienmarchal committed:
Merge remote-tracking branch 'upstream/master' into pr-replace-long-vec-alias-with-shortnames

damienmarchal committed:
[Sofa.Core] Add deprecated message and use VecX inplace of VecXd

damienmarchal committed:
[all] Remove commented code containing reference to sofa::type::vector

damienmarchal commented:
as requested, now using vecX instead of vecXd and has a deprecation workflow in place with: 
SOFA_ATTRIBUTE_DEPRECATED__VECTOR()

damienmarchal commented:
[ci-build]

damienmarchal committed:
Merge remote-tracking branch 'upstream/master' into pr-replace-long-vec-alias-with-shortnames

damienmarchal committed:
[all] Replace type::Vector3 with ::type::Vec3

damienmarchal committed:
[Sofa.Component.Collision] Remove commented code in IncrSAP.cpp

damienmarchal committed:
Merge remote-tracking branch 'cristal/pr-replace-long-vec-alias-with-shortnames' into pr-replace-long-vec-alias-with-shortnames

damienmarchal committed:
Merge remote-tracking branch 'upstream/master' into pr-replace-long-vec-alias-with-shortnames

# Conflicts:
#	Sofa/Component/Collision/Response/Mapper/tests/BaryMapper_test.cpp
#	Sofa/Component/Constraint/Projective/src/sofa/component/constraint/projective/AttachConstraint.cpp
#	Sofa/Component/Engine/Transform/src/sofa/component/engine/transform/TransformEngine.inl
#	Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/GridMeshCreator.cpp
#	Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/StringMeshCreator.cpp
#	Sofa/Component/IO/Mesh/src/sofa/component/io/mesh/VoxelGridLoader.cpp
#	Sofa/Component/Mapping/Linear/src/sofa/component/mapping/linear/BeamLinearMapping.inl
#	Sofa/Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/TriangularFEMForceFieldOptim.inl
#	Sofa/Component/Topology/Container/Dynamic/src/sofa/component/topology/container/dynamic/DynamicSparseGridTopologyContainer.cpp
#	Sofa/Component/Topology/Container/Grid/tests/RegularGridTopology_test.cpp
#	Sofa/Component/Visual/src/sofa/component/visual/RecordedCamera.cpp
#	Sofa/GL/Component/Shader/src/sofa/gl/component/shader/Light.cpp
#	Sofa/GL/src/sofa/gl/DrawToolGL.cpp
#	Sofa/GUI/Qt/src/sofa/gui/qt/viewer/qgl/QtGLViewer.cpp
#	Sofa/GUI/Qt/src/sofa/gui/qt/viewer/qt/QtViewer.cpp
#	Sofa/framework/Config/src/sofa/config.h.in
#	Sofa/framework/Core/src/sofa/core/behavior/BaseMechanicalState.h
#	Sofa/framework/Core/src/sofa/core/behavior/Mass.inl

damienmarchal commented:
[ci-build]

damienmarchal committed:
[all] Replace other type,

damienmarchal committed:
Merge remote-tracking branch 'upstream/master' into pr-replace-long-vec-alias-with-shortnames

# Conflicts:
#	Sofa/Component/Collision/Detection/Intersection/src/sofa/component/collision/detection/intersection/LocalMinDistance.cpp
#	Sofa/Component/Collision/Detection/Intersection/src/sofa/component/collision/detection/intersection/MeshDiscreteIntersection.cpp
#	Sofa/Component/Collision/Detection/Intersection/src/sofa/component/collision/detection/intersection/MeshMinProximityIntersection.cpp
#	Sofa/Component/Collision/Detection/Intersection/src/sofa/component/collision/detection/intersection/MeshMinProximityIntersection.h
#	Sofa/Component/Collision/Detection/Intersection/src/sofa/component/collision/detection/intersection/MeshNewProximityIntersection.cpp
#	Sofa/Component/Collision/Geometry/src/sofa/component/collision/geometry/LineModel.inl
#	Sofa/Component/Collision/Geometry/src/sofa/component/collision/geometry/PointModel.inl
#	Sofa/Component/Collision/Geometry/src/sofa/component/collision/geometry/TriangleModel.inl

damienmarchal committed:
Merge remote-tracking branch 'upstream/master' into pr-replace-long-vec-alias-with-shortnames

damienmarchal committed:
[Sofa.Core] Remove getValue() call from  notMuted()

The notMuted() method is called in msg_info(). When there is
msg_info() in inner-loop, this can induce a lot of overhead.

In this commit we refactor the implementation to reduce the overhead.

damienmarchal committed:
[all] More conversion from Vector3 to Vec3.

damienmarchal commented:
[ci-build][with-scene-tests][with-regression-tests]

damienmarchal committed:
[all] Change few things to make compilation with float only working.

damienmarchal committed:
Merge remote-tracking branch 'upstream/master' into pr-replace-long-vec-alias-with-shortnames

damienmarchal committed:
Revert "[Sofa.Core] Remove getValue() call from  notMuted()"

This reverts commit 1c5162381bcc9fa617f99bdda77b8ed02ee111fd.

damienmarchal commented on Sofa/Component/Constraint/Lagrangian/Solver/src/sofa/component/constraint/lagrangian/solver/GenericConstraintSolver.cpp:
```diff
@@ -393,7 +393,7 @@ void GenericConstraintSolver::buildSystem_matrixAssembly(const core::ConstraintP
     msg_info() << " computeCompliance_done "  ;
 }
 
-void GenericConstraintSolver::rebuildSystem(SReal massFactor, SReal forceFactor)
```
Yes ! 

It was a mistake to pass it to SReal because this method is overriding one with the following signature:  ConstraintSolver::rebuildSystem(double, double).

damienmarchal commented on Sofa/framework/Core/src/sofa/core/objectmodel/Base.cpp:
```diff
@@ -69,13 +69,6 @@ Base::Base()
     }, {&d_componentState});
-
-    /// name change => component state update
-    addUpdateCallback("muted", {&f_printLog}, [this](const DataTracker&){
```
Aren't you making a diff between the commit ? 
This line was added by mistake in a commit of the PR and then reverted.

damienmarchal committed:
Merge remote-tracking branch 'upstream/master' into pr-replace-long-vec-alias-with-shortnames

# Conflicts:
#	Sofa/Component/Collision/Geometry/src/sofa/component/collision/geometry/LineModel.inl

damienmarchal committed:
Merge remote-tracking branch 'upstream/master' into pr-replace-long-vec-alias-with-shortnames

# Conflicts:
#	Sofa/Component/SolidMechanics/FEM/Elastic/src/sofa/component/solidmechanics/fem/elastic/HexahedronFEMForceField.inl

fredroy reviewed (approved):
Looks good to me ! and marking it "breaking"

fredroy commented:
Last check : [ci-build][with-all-tests]

damienmarchal commented:
Thanks Fred !
